### DISCUSSION TITLE
Text font size resizing on scaling

### DISCUSSION THREAD

**Discussion by SirPm** [fabricjs/fabric.js#7541] [Q&A] [answered]
Hi, everyone, please does anyone have an idea on how to prevent fabric object Textbox from resizing text when scaled.

For example while styling, I change the fontSize of the text to 36 for example, but when I resize and scale up or down using the controls, it either shrinks the text or enlarges it, but I want the text to still remain the same size and only change the bounding box.

Thanks in advance for any help regarding this

**Comment by asturur** [comment, marked as answer]
ho sirPm, there are 2 ways here.
One is using the scale event to reduce font size ( but that is probably not what you want )
The other is to take the changeWidth control, create a specular changeHeight control handler, and combine them in the corners of the textbox.

This is confusing i assume.

Please have a look at the custom control api, start to setup something and tell us where you are stuck at.

**Reply by SirPm** [reply]
After searching on stackoverflow, I eventually saw a solution that worked for me

canvas.on("object:scaling", function (obj: IFabricEvent) {
				if (obj.target && obj.target.id && obj.target.height && obj.target.scaleY) {
					if (obj.target.id.includes("txt")) {
						let lastHeight: number;

						const updateTextSize = () => {
							if (obj.target) {
								if (obj.target.height && obj.target.scaleY) {
									lastHeight = obj.target.height * obj.target.scaleY;
								}

								obj.target.set({
									height: lastHeight || obj.target.height,
									scaleY: 1,
								});

								canvas.renderAll();
							}
						};

						updateTextSize();
					}
				}
			});

So this was what I found out as a solution and it worked for me
